crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -143,93 +297,201 @@ impl<F, T>
+                    attempt_status: None,
+                    connector_transaction_id: None,
+                    status_code: item.http_code,
+                }),
+                ..item.data
+            }),
+            DataTransRefundsResponse::Success(response) => Ok(Self {
+                response: Ok(types::RefundsResponseData {
+                    connector_refund_id: response.transaction_id,
+                    refund_status: enums::RefundStatus::Success,
+                }),
+                ..item.data
+            }),
+        }
+    }
+}
 
-#[allow(dead_code)]
-#[derive(Debug, Serialize, Default, Deserialize, Clone)]
-pub enum RefundStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
+impl TryFrom<types::RefundsResponseRouterData<api::RSync, DataTransSyncResponse>>
+    for types::RefundsRouterData<api::RSync>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::RSync, DataTransSyncResponse>,
+    ) -> Result<Self, Self::Error> {
+        let response = match item.response {
+            DataTransSyncResponse::Error(error) => Err(types::ErrorResponse {
+                code: error.code.unwrap_or(consts::NO_ERROR_CODE.to_string()),
+                message: error
+                    .message
+                    .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                reason: None,
+                attempt_status: None,
+                connector_transaction_id: None,
+                status_code: item.http_code,
+            }),
+            DataTransSyncResponse::Response(response) => Ok(types::RefundsResponseData {
+                connector_refund_id: response.transaction_id.to_string(),
+                refund_status: enums::RefundStatus::from(response),
+            }),
+        };
+        Ok(Self {
+            response,
+            ..item.data
+        })
+    }
 }
 
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
-        match item {
-            RefundStatus::Succeeded => Self::Success,
-            RefundStatus::Failed => Self::Failure,
-            RefundStatus::Processing => Self::Pending,
-            //TODO: Review mapping
+impl TryFrom<types::PaymentsSyncResponseRouterData<DataTransSyncResponse>>
+    for types::PaymentsSyncRouterData
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::PaymentsSyncResponseRouterData<DataTransSyncResponse>,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            DataTransSyncResponse::Error(error) => {
+                let response = Err(types::ErrorResponse {
+                    code: error.code.unwrap_or(consts::NO_ERROR_CODE.to_string()),
+                    message: error
+                        .message
+                        .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                    reason: None,
```

Comment:
Populate `error.message` in `reason` field too. This is required because Hyperswitch displays this reason on dashboard only.